### PR TITLE
Add support for sensitive options and implement -D -S switch

### PR DESCRIPTION
**Pull request by gahr**
Issue: #236

### CODE REVIEW

**Comment by gahr** [comment]
TBD 
* rename `R_SENSITIVE` so it's not in the `R_*` namespace? If so, how to ensure bitwise flaggability?
* what exactly to print? `'***'`, `***`, `********`, `<***>`; should this be a config option? Overkill imho..

**Comment by guyzmo** [comment]
another question:

- shall we go for `-C` or `-D -S` like [I implemented](https://github.com/neomutt/neomutt/pull/300/files#diff-2045016cb90d1e65d71c2407a2570927R306)?

File: init.h [hunk 1]
```diff
@@ -68,6 +68,7 @@
 #define R_SIDEBAR       (1<<7)  /* redraw the sidebar */
 #define R_BOTH		(R_INDEX | R_PAGER)
 #define R_RESORT_BOTH	(R_RESORT | R_RESORT_SUB)
+#define R_SENSITIVE     (1<<8)  /* hide in mutt -C */
```
Comment (guyzmo) [inline init.h]: cf earlier comments, choose a prefix to properly namespace this, like `Q_` for qualifier…

**Comment by gahr** [comment]
Yeah I'll take care of the docs as soon as we decide what option to implement ;)

**Comment by gahr** [comment]
After much thought (...) I agree on the superiority of "-D -S" as opposed to "-C" :)

**Comment by guyzmo** [comment]
one last thing, you need to update the [xml documentation](https://github.com/neomutt/neomutt/blob/neomutt/doc/manual.xml.head#L14147)

**Pull request closed without merging**